### DISCUSSION TITLE
nameof and method parameter names

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#3191] [Language Ideas]
It would be great if NameOf could reference parameters from outside the scope:
````
public void Method1(){
  var Value = nameof(Method2.Param2);
}

public void Method2(object sender, object Param2){

}

````

**Comment by canton7** [comment]
Duplicate of #373?

**Comment by mpawelski** [comment]
Looks like #373 is less general and talks only about nameof of methods parametes used in attribute of that method. 
And probably we'll get this because it will nicely work with attributes like `NotNullIfNotNull`.

However having more general feature that adds possibility to use nameof for *any* methods' parameters like with `nameof(Method2.Param2)` syntax will be still very useful. 

I think there are many frameworks where stuff is more "[stringly typed](https://wiki.c2.com/?StringlyTyped)". For example in ASP.NET helpers for urls you need to provide the same name of parameter as your action method to generate proper url with working model binding. Just using `nameof(MyController.MyAction.param1)` would make it clear why we need to use "param1" string.
